x-pack/plugins/ml/public/application/components/shared/add_inference_pipeline_horizontal_steps.tsx

```diff
@@ -96,23 +80,60 @@ export const AddInferencePipelineHorizontalSteps: FC<Props> = memo(
         ),
       },
     ];
+
+    if (hasProcessorStep === true) {
+      navSteps.splice(1, 0, {
+        // Processor configuration
+        onClick: () => {
+          if (!isDetailsStepValid) return;
+          setStep(ADD_INFERENCE_PIPELINE_STEPS.CONFIGURE_PROCESSOR);
+        },
+        status:
+          isDetailsStepValid && isConfigureProcessorStepValid && currentStepIndex > 1
+            ? 'complete'
+            : 'incomplete',
+        title: i18n.translate(
+          'xpack.ml.inferencePipeline.content.indices.transforms.addInferencePipelineModal.steps.configureProcessor.title',
+          {
+            defaultMessage: 'Configure processor',
+          }
+        ),
+      });
+    }
+    let DETAILS_INDEX: number;
+    let CONFIGURE_INDEX: number | undefined;
+    let ON_FAILURE_INDEX: number;
+    let TEST_INDEX: number;
+    let CREATE_INDEX: number;
+
+    if (hasProcessorStep) {
+      [DETAILS_INDEX, CONFIGURE_INDEX, ON_FAILURE_INDEX, TEST_INDEX, CREATE_INDEX] = [
+        0, 1, 2, 3, 4, 5,
+      ];
+    } else {
+      [DETAILS_INDEX, ON_FAILURE_INDEX, TEST_INDEX, CREATE_INDEX] = [0, 1, 2, 3, 4];
+    }
+
     switch (step) {
       case ADD_INFERENCE_PIPELINE_STEPS.DETAILS:
-        navSteps[0].status = 'current';
+        navSteps[DETAILS_INDEX].status = 'current';
         break;
       case ADD_INFERENCE_PIPELINE_STEPS.CONFIGURE_PROCESSOR:
-        navSteps[1].status = 'current';
+        if (CONFIGURE_INDEX !== undefined) {
+          navSteps[CONFIGURE_INDEX].status = 'current';
+        }
         break;
       case ADD_INFERENCE_PIPELINE_STEPS.ON_FAILURE:
-        navSteps[2].status = 'current';
+        navSteps[ON_FAILURE_INDEX].status = 'current';
         break;
       case ADD_INFERENCE_PIPELINE_STEPS.TEST:
-        navSteps[3].status = 'current';
+        navSteps[TEST_INDEX].status = 'current';
         break;
       case ADD_INFERENCE_PIPELINE_STEPS.CREATE:
-        navSteps[4].status = 'current';
+        navSteps[CREATE_INDEX].status = 'current';
         break;
     }
+
     return <EuiStepsHorizontal steps={navSteps} size="s" />;
```

Comment:
Should we disable the first three steps once the pipeline has been created (applies to the DFA wizard too). Currently you can go back and add some edits after creation but there is no way to apply these edits as the 'Create' button is no longer available in step 4.

Reply:
Disabled in 348a4d0939c46bf99383430b00471b470058aafa 
This is a shared component so it will also work in the DFA wizard.